### ISSUE TITLE
Textdomain should not use a variable

### ISSUE THREAD

**Issue by robneu** [simmerwp/simmer#101]
Right now the plugin's textdomain is set using a public property of the `Simmer` class which references a constant of the same class. This can cause problems when translating the plugin and should be replaced with a hard-coded string.

There are quite a few reasons for this, but it's already been explained better by others:

https://ulrich.pogson.ch/string-text-domain-must
http://ottopress.com/tag/internationalization/
https://markjaquith.wordpress.com/2011/10/06/translating-wordpress-plugins-and-themes-dont-get-clever/

The short version is that you shouldn't use PHP code to set textdomains in plugins or themes. This will be even more of an issue if you'd like to use the WordPress.org hosted language packs.

**Comment by laralfield** [comment]
👍 Thanks for the detailed explanation here.
